File: src/main/java/com/fateironist/jawf/model/Agent.java
package com.fateironist.jawf.model;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * Agent 实体类。
 * <p>
 * 表示一个可配置的 AI Agent，包含名称、模型、并行能力等属性。
 * 用于持久化存储 Agent 配置信息，支持后续扩展记忆、工具等能力。
 */
@Data
public class Agent {

    /** 自增主键 */
    private Long id;

    /** Agent 唯一标识（UUID 格式） */
    private String agentId;

    /** Agent 名称 */
    private String name;

    /** Agent 描述 */
    private String description;

    /** 默认模型标识（格式：厂商名_模型名，如 dashscope_deepseek-v4-flash） */
    private String defaultModel;

    /** 系统提示词 */
    private String systemPrompt;

    /** 是否支持并行执行 */
    private Boolean parallelEnabled;

    /** 最大并行数（parallelEnabled=true 时有效） */
    private Integer maxParallel;

    /** 最大重试次数 */
    private Integer maxRetry;

    /** 超时时间（秒） */
    private Integer timeoutSeconds;

    /** Agent 类型（如 llm、tool、workflow 等） */
    private String agentType;

    /** 扩展配置（JSON 格式） */
    private String configJson;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;

    /** 是否启用 */
    private Boolean enabled;
}

File: front/src/api/agent.ts
import request from './request'

export interface Agent {
  id: number
  agentId: string
  name: string
  description: string
  defaultModel: string
  systemPrompt: string
  parallelEnabled: boolean
  maxParallel: number
  maxRetry: number
  timeoutSeconds: number
  agentType: string
  configJson: string
  createdAt: string
  updatedAt: string
  enabled: boolean
}

export interface CreateAgentRequest {
  name: string
  description?: string
  defaultModel: string
  systemPrompt?: string
  parallelEnabled?: boolean
  maxParallel?: number
  agentType?: string
}

export interface UpdateAgentRequest {
  name?: string
  description?: string
  defaultModel?: string
  systemPrompt?: string
  parallelEnabled?: boolean
  maxParallel?: number
  agentType?: string
}

// 获取所有启用的 Agent
export function listAgents() {
  return request.get<any, Agent[]>('/agents')
}

// 根据 agentId 获取 Agent
export function getAgent(agentId: string) {
  return request.get<any, Agent>(`/agents/${agentId}`)
}

// 创建 Agent
export function createAgent(data: CreateAgentRequest) {
  return request.post<any, Agent>('/agents', data)
}

// 更新 Agent
export function updateAgent(agentId: string, data: UpdateAgentRequest) {
  return request.put<any, Agent>(`/agents/${agentId}`, data)
}

// 删除 Agent
export function deleteAgent(agentId: string) {
  return request.delete(`/agents/${agentId}`)
}

// 启用/禁用 Agent
export function setAgentEnabled(agentId: string, enabled: boolean) {
  return request.patch(`/agents/${agentId}/enabled`, { enabled })
}
